### PR TITLE
`azurerm_sentinel_log_analytics_workspace_onboarding`: support `workspace_id`, deprecate `resource_group_name` and `workspace_name`

### PR DESCRIPTION
**Pull request by ziyeqf**
Screenshot 2023-02-27 at 14 42 12
the failed ones are because of `InvalidLicense`

Acctests may fail with is not onboarded 
```
Error: checking for presence of existing Data Connector: (Name "accTestDC-230224022526508153" / Workspace Name "acctestLAW-230224022526508153" / Resource Group "acctestRG-sentinel-230224022526508153"): securityinsight.DataConnectorsClient#Get: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="BadRequest" Message="{\"error\":{\"code\":\"BadRequest\",\"message\":\"Workspace 'acctestLAW-230224022526508153' is not onboarded to Microsoft Sentinel.\"}}"
```
change from `azurerm_log_analytics_solution` to `azurerm_sentinel_log_analytics_workspace_onboarding`. 

`azurerm_sentinel_alert_rule_fusion` is excluded, the service is installing `Advanced Multistage Attack Detection` by default in `onboarding` API, the only one fusion rule. It's also installed by default by Portal.  May update its test case in another PR

### CODE REVIEW

File: internal/services/sentinel/sentinel_alert_rule_machine_learning_behavior_analytics_test.go [hunk 1]
```diff
@@ -137,26 +141,34 @@ func (r SentinelAlertRuleMLBehaviorAnalyticsResource) complete(data acceptance.T
...
 }
 
 resource "azurerm_sentinel_alert_rule_machine_learning_behavior_analytics" "test2" {
   name                       = "acctest-SentinelAlertRule-MLBehaviorAnalytics-2-%d"
-  log_analytics_workspace_id = azurerm_log_analytics_solution.test.workspace_resource_id
+  log_analytics_workspace_id = azurerm_log_analytics_workspace.test.id
   alert_rule_template_guid   = data.azurerm_sentinel_alert_rule_template.test2.name
   enabled                    = false
+
+  depends_on = [azurerm_sentinel_log_analytics_workspace_onboarding.test]
```
Comment (katbyte) [inline internal/services/sentinel/sentinel_alert_rule_machine_learning_behavior_analytics_test.go:171]: could you not do:

```suggestion
  log_analytics_workspace_id = azurerm_sentinel_log_analytics_workspace_onboarding.test.workspace_id
  alert_rule_template_guid   = data.azurerm_sentinel_alert_rule_template.test2.name
  enabled                    = false
```

File: internal/services/sentinel/sentinel_alert_rule_machine_learning_behavior_analytics_test.go [hunk 2]
```diff
@@ -191,17 +203,9 @@ resource "azurerm_log_analytics_workspace" "test" {
...
-  workspace_resource_id = azurerm_log_analytics_workspace.test.id
-  workspace_name        = azurerm_log_analytics_workspace.test.name
-
-  plan {
-    publisher = "Microsoft"
-    product   = "OMSGallery/SecurityInsights"
-  }
+resource "azurerm_sentinel_log_analytics_workspace_onboarding" "test" {
+  resource_group_name = azurerm_resource_group.test.name
+  workspace_name      = azurerm_log_analytics_workspace.test.name
```
Comment (katbyte) [inline internal/services/sentinel/sentinel_alert_rule_machine_learning_behavior_analytics_test.go:208]: and we can (should?) update this to simple take an id
```suggestion
resource "azurerm_sentinel_log_analytics_workspace_onboarding" "test" {
  workspace_id      = azurerm_log_analytics_workspace.test.id
```

**Comment by ziyeqf** [comment]
convert to draft to wait for test result

**Comment by ziyeqf** [comment]
Screenshot 2023-02-28 at 18 50 36

**Review by katbyte — APPROVED** [review]
LGTM 👽

**Pull request merged**